### PR TITLE
Cli framework choice

### PR DESCRIPTION
**Pull request by Keraito**
Issue: Part of #1108

Added a `--type -t` flag to the `storybook init` command so users can manually choose which framework/template of Storybook to install. It takes a `type` parameter which (for now) needs to be an entry in `lib/cli/lib/project_types.js`. It also opens an inquirer when nothing is detected so that the user can choose one anyways if they like.

## Example of providing type flag
storybook-type-install

## Example of inquirer when undetected SB
storybook-undetected-install

All the options in the inquirer and the accepted `types` are extracted from `lib/cli/lib/project_types.js`. This way, when ppl add new frameworks to SB they don't have to make changes in the CLI either (f.e. the confusion in #4161) and it will be auto included. 

Some issues/remarks:
- Do people agree with adding this as a flag in such a way?
- The keys/values in the `project_types.js` aren't really that descriptive. It's most likely that this will only be used on projects that are not compatible with SB, so most projects will be auto-detected anyways. But it might be vague what the difference is f.e. between `REACT_PROJECT`, `REACT`, and `REACT_SCRIPTS`.
- Probably nice to add an option to the inquirer list that their project framework/type is not included, and then provide information like SB GH issues page if they really want it for their project?
- Not sure how to efficiently test this.
- The CLI folder is kinda getting cluttered, probably needs some structure soon 😅 .

### CODE REVIEW

File: lib/cli/lib/initiate.js [hunk 1]
```diff
@@ -176,17 +157,83 @@ export default function(options, pkg) {
           "Please make sure you are running the `storybook init` command in your project's root directory."
         );
         paddedLog(
-          'You can also install storybook for plain HTML snippets with `storybook init --html` or follow some of the slow start guides: https://storybook.js.org/basics/slow-start-guide/'
+          'You can also install storybook for plain HTML snippets with `storybook init --type html` or follow some of the slow start guides: https://storybook.js.org/basics/slow-start-guide/'
```
Comment (Hypnosphi) [inline lib/cli/lib/initiate.js]: I think those lines should be changed to something like "you can specify the project type explixitly via ..."

**Comment by libetl** [comment]
It looks good to me, but do can you write a unit test ?

**Review by libetl — APPROVED** [review]
But I would advise to write an unit test

**Comment by igor-dv** [comment]
@Keraito 

I've found one more thing that should be changed:

image

**Comment by ndelangen** [comment]
```
 👑
 😎
✌👕⁣✌
 👖
 👟👟
```

this is cool @Keraito !

**Comment by Keraito** [comment]
@libetl Will see what I can do. As stated in the first comment, I'm not quite sure how to write a meaningful (unit) and efficient test for this. But I will see what I can do.

@igor-dv Thanks! Missed it somehow.

@ndelangen That actually took me ages to read correctly 🤔 More to come!

**Comment by tomByrer** [comment]
I like the 'inquirer'/ framework chooser.

**Comment by Keraito** [comment]
Updated with some unit tests.

**Review by ndelangen — APPROVED** [review]

**Comment by ndelangen** [comment]
Super awesome @Keraito 💪

**Comment by igor-dv** [comment]
@Keraito, I think you can fix the conflict and merge (don't forget to address my comment in some PR, though).

**Comment by ndelangen** [comment]
Which screenshot?

**Comment by Keraito** [comment]
https://github.com/storybooks/storybook/blob/master/lib/cli/docs/getstorybook.png @ndelangen

**Comment by igor-dv** [comment]
Doesn't git history show something? I think you can print-screen it from whatever tool you want =)

**Comment by pksunkara** [comment]
@Keraito https://github.com/marionebl/svg-term. I used it for https://github.com/storybooks/vue-cli-plugin-storybook

**Review by wuweiweiwu — APPROVED** [review]
LGTM! NICE WORK 💯

**Comment by Keraito** [comment]
Resolved the merge conflicts

File: lib/cli/lib/project_types.js [hunk 2]
```diff
@@ -19,3 +19,25 @@ export default {
...
+
+export const supportedFrameworks = [
+  'react',
+  'react-native',
+  'vue',
+  'angular',
+  'polymer',
+  'mithril',
+  'riot',
+];
```
Comment (shilman) [inline lib/cli/lib/project_types.js]:
```
ember
marko
meteor?
```

**Review by shilman — APPROVED** [review]
Super improvement!!! 🎖

**Comment by shilman** [comment]
@Keraito felt guilty about creating more merge conflicts so I resolved them for yuo (hopefully correctly)!

**Comment by libetl** [comment]
Good job, I think everything is ready, don't you ?

**Comment by ndelangen** [comment]
@shilman I assume you tested manually? If so go ahead and merge

**Comment by ndelangen** [comment]
This is great

File: lib/cli/lib/project_types.js (hunk 2 above)
Comment (Keraito) [reply lib/cli/lib/project_types.js]: Hmmm I just extracted this array to another file. Ember support wasn't present yet when I made this PR, but marko en meteor def should have been. I don't see them in the removed code either tho 😱 will add them I'm a few hours

**Comment by Keraito** [comment]
@shilman it wasn't that big of a deal, but thank you very much! 🙇‍♂️

**Comment by Hypnosphi** [comment]
Looks like this is the only thing left: https://github.com/storybooks/storybook/pull/4184/files#r224264956

**Pull request merged**